### ISSUE TITLE
Where is Keras 2 's channels?

### ISSUE THREAD

**Issue by gladuo** [fchollet/keras#6250]
I once used keras 1 (maybe 1.0.5) for multi-category classification. And my input in CNN is (n, 1, 24, 113) and 113 is channel numbers, and kernel size is (1, 5).
code like:

    X_train = X_train.reshape((-1, 1, SLIDING_WINDOW_LENGTH, NUM_SENSOR_CHANNELS))
    X_test = X_test.reshape((-1, 1, SLIDING_WINDOW_LENGTH, NUM_SENSOR_CHANNELS))
    
    # network
    inputs = Input(shape=(1, SLIDING_WINDOW_LENGTH, NUM_SENSOR_CHANNELS))
    conv1 = ELU()(Convolution2D(NUM_FILTERS, FILTER_SIZE, 1, border_mode='valid', init='normal', activation='relu')(inputs))
    conv2 = ELU()(Convolution2D(NUM_FILTERS, FILTER_SIZE, 1, border_mode='valid', init='normal', activation='relu')(conv1))
    conv3 = ELU()(Convolution2D(NUM_FILTERS, FILTER_SIZE, 1, border_mode='valid', init='normal', activation='relu')(conv2))
    conv4 = ELU()(Convolution2D(NUM_FILTERS, FILTER_SIZE, 1, border_mode='valid', init='normal', activation='relu')(conv3))
    reshape1 = Reshape((8, NUM_FILTERS * NUM_SENSOR_CHANNELS))(conv4)
    gru1 = GRU(NUM_UNITS_LSTM, return_sequences=True, consume_less='mem')(reshape1)
    gru2 = GRU(NUM_UNITS_LSTM, return_sequences=False, consume_less='mem')(gru1)
    outputs = Dense(NUM_CLASSES, activation='softmax')(gru2)

    # Hardcoded number of sensor channels employed in the OPPORTUNITY challenge
    NUM_SENSOR_CHANNELS = 113
    
    # Hardcoded number of classes in the gesture recognition problem
    NUM_CLASSES = 18
    
    # Hardcoded length of the sliding window mechanism employed to segment the data
    SLIDING_WINDOW_LENGTH = 24
    
    # Length of the input sequence after convolutional operations
    FINAL_SEQUENCE_LENGTH = 8
    
    # Hardcoded step of the sliding window mechanism employed to segment the data
    SLIDING_WINDOW_STEP = 12
    
    # Batch Size
    BATCH_SIZE = 100
    
    # Number filters convolutional layers
    NUM_FILTERS = 64
    
    # Size filters convolutional layers
    FILTER_SIZE = 5
    
    # Number of unit in the long short-term recurrent layers
    NUM_UNITS_LSTM = 128

And these days I switched keras to keras 2. and the networks did not change. And my code like:

    X_train = X_train.reshape((-1, 1, SLIDING_WINDOW_LENGTH, NUM_SENSOR_CHANNELS))
    X_test = X_test.reshape((-1, 1, SLIDING_WINDOW_LENGTH, NUM_SENSOR_CHANNELS))
    
    # network
    inputs = Input(shape=(1, SLIDING_WINDOW_LENGTH, NUM_SENSOR_CHANNELS))
    conv1 = ELU()(
        Conv2D(filters=NUM_FILTERS, kernel_size=(1, FILTER_SIZE), strides=(1, 1), padding='valid', activation='relu',
               kernel_initializer='normal', data_format='channels_last')(inputs))
    conv2 = ELU()(
        Conv2D(filters=NUM_FILTERS, kernel_size=(1, FILTER_SIZE), strides=(1, 1), padding='valid', activation='relu',
               kernel_initializer='normal', data_format='channels_last')(conv1))
    conv3 = ELU()(
        Conv2D(filters=NUM_FILTERS, kernel_size=(1, FILTER_SIZE), strides=(1, 1), padding='valid', activation='relu',
               kernel_initializer='normal', data_format='channels_last')(conv2))
    conv4 = ELU()(
        Conv2D(filters=NUM_FILTERS, kernel_size=(1, FILTER_SIZE), strides=(1, 1), padding='valid', activation='relu',
               kernel_initializer='normal', data_format='channels_last')(conv3))
    # permute1 = Permute((2, 1, 3))(conv4)
    reshape1 = Reshape((SLIDING_WINDOW_LENGTH - (FILTER_SIZE - 1) * 4, NUM_FILTERS * 1))(conv4)  # 4 for 4 convs
    gru1 = GRU(NUM_UNITS_LSTM, return_sequences=True, implementation=0)(reshape1)
    gru2 = GRU(NUM_UNITS_LSTM, return_sequences=False, implementation=0)(gru1)  # implementation=2 for GPU
    outputs = Dense(NUM_CLASSES, activation='softmax')(gru2)

and the speed seems faster but the shape is strange since I didn't know where is my channels ?

Is there anything wrong with my code and could someone help ? THX

**Comment by joelthchao** [comment]
Check this [FAQ](https://keras.io/getting-started/faq/#where-is-the-keras-configuration-filed-stored). Let Keras automatically configs your channels.
